extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -683,16 +685,18 @@ private static <DATA_TYPE> boolean tryEncodeDictionary(
                                 hasNulls = true;
                             } else {
                                 if (keyCount == encodedKeys.length) {
-                                    if (keyCount >= maxKeys) {
-                                        throw new DictionarySizeExceededException(
-                                                "Dictionary maximum size exceeded for " + columnDefinition.getName());
-                                    }
-
+                                    // Copy into an array of double the size with upper limit at maxKeys
                                     encodedKeys = Arrays.copyOf(encodedKeys, (int) Math.min(keyCount * 2L, maxKeys));
                                 }
                                 encodedKeys[keyCount] = Binary.fromString(key);
                                 dictionaryPos = keyCount;
+                                dictSize += encodedKeys[keyCount].length();
                                 keyCount++;
+                                if ((keyCount >= maxKeys) || (dictSize >= maxDictSize)) {
+                                    throw new DictionarySizeExceededException(
```

Comment:
We are discarding a lot of work assembling a dictionary and encoded chunks when we error and exit this function.  Our own reader can read hybrid columns (both RLE_DICTIONARY and PLAIN). If this does not violate an acceptable standard, I think we should try to leverage this work.

Reply:
Thanks for pointing this out, I have noted it as a comment on #946 and added a comment in the code.